core/conn/odb/src/odb.c

```diff
@@ -5313,7 +5313,7 @@ static void etabadd(char type, char *run, int id)
                     }
                 }
                 if ( etab[no].type == 'e' ) { /* name & create output file */
-                    for ( i = j = 0; etab[no].tgt[i] && i < sizeof(buff); i++ ) {
+                    for ( i = j = 0; i < sizeof(buff) && etab[no].tgt[i]; i++ ) {
```

Comment:
I don't really get this fully understand.
If i is out of bound for arry tgt, why this change will prevent the tag[i] to be executed?

Reply:
Most of time, they are equivalent.  There are exception in a special case 
Most of compiler judge the condition from left to right, so if the tgt is at the end of memory, and the offset i is equal to sizeof(buff), the access to memory goes beyond memory.
This will make a coredump, and cann't be repeatable.

Reply:
I tried to make sense of this too. It looks like etab[no].tgt is populated in function parseopt from some string buffer; I can't tell if it is buff (which is a static variable declared as a char[256]) or something else. My suspicion is that the test "i < sizeof(buff)" has nothing to do with etab[no].tgt but rather some other buffer being copied to. In any case, reversing the order of these tests seems harmless.